org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/TychoPackagingsConfigurator.java

```diff
@@ -0,0 +1,131 @@
+        IProject project = request.mavenProjectFacade().getProject();
+
+        String packaging = mavenProject.getPackaging();
+        if ("eclipse-plugin".equals(packaging) || "eclipse-test-plugin".equals(packaging)) {
+            PDEProjectHelper.configurePDEBundleProject(project, mavenProject, monitor);
+            applyDsConfiguration(request, monitor);
+        } else if ("eclipse-feature".equals(packaging)) {
+            // see org.eclipse.pde.internal.ui.wizards.feature.AbstractCreateFeatureOperation
+            if (!project.hasNature(PDE.FEATURE_NATURE)) {
+                CoreUtility.addNatureToProject(project, PDE.FEATURE_NATURE, monitor);
+            }
+        }
+    }
+
+    void applyDsConfiguration(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        List<MojoExecution> mojoExecutions = getTychoDsPluginMojoExecutions(request, monitor);
+        if (mojoExecutions.isEmpty()) {
+            return;
+        }
+        if (mojoExecutions.size() > 1) {
+            log.error("Found more than one execution for plugin {}:{} and goal {}, only consider configuration of first one", TYCHO_GROUP_ID, TYCHO_DS_PLUGIN_ARTIFACT_ID, GOAL_DECLARATIVE_SERVICES);
+        }
+        // first mojo execution is relevant
+        Xpp3Dom dom = mojoExecutions.get(0).getConfiguration();
+        // apply PDE configuration for DS
+        IEclipsePreferences prefs = new ProjectScope(request.mavenProjectFacade().getProject())
+                .getNode(org.eclipse.pde.ds.internal.annotations.Activator.PLUGIN_ID);
+        Xpp3Dom dsEnabled = dom.getChild("enabled");
+        final boolean isDsEnabled;
+        if (dsEnabled != null && !dsEnabled.getValue().isEmpty()) {
+            isDsEnabled = Boolean.valueOf(dsEnabled.getValue());
+            prefs.putBoolean(org.eclipse.pde.ds.internal.annotations.Activator.PREF_ENABLED, isDsEnabled);
+        } else {
+            isDsEnabled = false;
+        }
+        if (isDsEnabled) {
+            Xpp3Dom dsVersion = dom.getChild("dsVersion");
+            if (containsExplicitConfigurationValue(dsVersion)) {
+                DSAnnotationVersion version = parseVersion(dsVersion.getValue());
+                if (version != null) {
+                    prefs.put(org.eclipse.pde.ds.internal.annotations.Activator.PREF_SPEC_VERSION, version.name());
+                }
+            }
+            Xpp3Dom path = dom.getChild("path");
+            if (containsExplicitConfigurationValue(path)) {
+                prefs.put(org.eclipse.pde.ds.internal.annotations.Activator.PREF_PATH, path.getValue());
+            }
+        }
+    }
+    
+    private boolean containsExplicitConfigurationValue(Xpp3Dom config) {
+        if (config == null) {
+            return false;
+        }
+        String value = config.getValue();
+        // check if value is a property name
+        if (value.startsWith("${")) {
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
		// check if value is a property name
		return config != null && !config.getValue().startsWith("${");
```
Although the goal isn't to make everything as compact as possible, I think in this case a one-liner is more readable here.

Reply:
Done in https://github.com/eclipse-m2e/m2e-core/pull/967/commits/b30f5013acbf8e7e38c8e6166de8e9ac876f278a.